### PR TITLE
[156276] enable to set local path as gear dependencies

### PR DESCRIPTION
**Pull request by s417-lama**
https://acsmine.tok.access-company.com/redmine/issues/156276

### CODE REVIEW

File: core/util/path.ex [hunk 1]
```diff
@@ -11,12 +11,12 @@ defmodule AntikytheraCore.Path do
...
-    defun antikythera_root_dir()  :: Path.t, do: @antikythera_root_dir
-    defun compiled_core_dir() :: Path.t, do: Path.join(antikythera_root_dir(), "releases")
+    defun antikythera_root_dir() :: Path.t, do: @antikythera_root_dir
+    defun compiled_core_dir()    :: Path.t, do: Path.join(antikythera_root_dir(), "releases")
   else
     # Use runtime application config
-    defun antikythera_root_dir()  :: Path.t, do: Application.fetch_env!(:antikythera, :antikythera_root_dir)
-    defun compiled_core_dir() :: Path.t do
+    defun antikythera_root_dir() :: Path.t, do: Application.fetch_env!(:antikythera, :antikythera_root_dir)
+    defun compiled_core_dir()    :: Path.t do
```
Comment (skirino) [inline core/util/path.ex]: Thank you for fixing these lines (mis-aligned by renaming)!

File: core/mix/ensure_gear_dependencies.ex [hunk 2]
```diff
@@ -13,22 +13,41 @@ defmodule Mix.Tasks.Compile.EnsureGearDependencies do
...
       Mix.raise("#{gear_name} is not a gear application! `gear_deps/0` may only contain gear dependencies.")
     end
   end
 
-  defp gear_dir(gear_name) do
-    cwd = File.cwd!()
+  defp gear_dir(gear_name, opts) do
+    case Keyword.fetch(opts, :path) do
+      {:ok, path} -> gear_dir_as_local_dep(path)
+      :error      -> gear_dir_as_remote_dep(gear_name)
```
Comment (skirino) [inline core/mix/ensure_gear_dependencies.ex]: `gear_dir_as_path_dep`, `gear_dir_as_non_path_dep`
could sound nicer for these functions in the sense that the term "path dependency" is used by official docs such as [`Mix.Tasks.Deps`](https://hexdocs.pm/mix/Mix.Tasks.Deps.html).

**Comment by ttokit** [comment]
LGTM!

**Comment by s417-lama** [comment]
@ymtszw I'm going to merge this PR right now. Please review this when you have enough time.

**Pull request merged**